### ISSUE TITLE
grafana_team: Unable to add team with space in name

### ISSUE THREAD

**Issue by rsirny** [ansible-collections/community.grafana#164]
##### SUMMARY
grafana_team: Unable to add team with space in name

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
grafana_team

##### ANSIBLE VERSION

```paste below
ansible 2.9.15
  config file = /mnt/c/myrepo/ansible.cfg
  configured module search path = ['/home/myuser/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.9.0 (default, Oct  6 2020, 00:00:00) [GCC 10.2.1 20200826 (Red Hat 10.2.1-3)]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

Fedora 33; Grafana 7.5.4

##### STEPS TO REPRODUCE
Run the following playbook.

The root cause is missing URL-friendly encoding in [get_team](https://github.com/ansible-collections/community.grafana/blob/e015c809f9baf92c1d7a8c20b17a684bfda7477a/plugins/modules/grafana_team.py#L240) function. It seems that similar [issue](https://github.com/ansible-collections/community.grafana/issues/78) has been reported and fixed for grafana_folder module.

```yaml
---
- hosts: localhost
  gather_facts: no
  become: no
  tasks:
  - name: Add teams
    community.grafana.grafana_team:
      grafana_url: "{{ grafana_base_uri }}"
      grafana_user: "{{ grafana_admin_user }}"
      grafana_password: "{{ grafana_admin_password }}"
      name: "Team With Space"
      email: ""
      state: present
```

##### EXPECTED RESULTS

Team "Team With Space" with space in name is created.

##### ACTUAL RESULTS
Add team step fails with message "Grafana Teams API answered with HTTP -1".

```paste below
PLAY [localhost] ************************************************************************************************************************************************************************************************
TASK [Add teams] ************************************************************************************************************************************************************************************************fatal: [localhost]: FAILED! => {"changed": false, "msg": "Grafana Teams API answered with HTTP -1"}

PLAY RECAP ******************************************************************************************************************************************************************************************************localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

**Comment by CWollinger** [comment]
Should be fixed with #169

**Comment by Nemental** [comment]
Can be closed - fixed by #169 and also tested by changes in #328
